apps/site/assets/ts/hooks/useVehiclesChannel.ts

```diff
@@ -0,0 +1,137 @@
+import { pick, keyBy } from "lodash";
+import deepEqual from "fast-deep-equal/react";
+import { Reducer } from "react";
+import { DirectionId } from "../__v3api";
+import useChannel from "./useChannel";
+import { CrowdingType } from "../schedule/components/__schedule";
+
+/**
+ * The format of a vehicle emitted by the VehiclesChannel
+ */
+export interface VehicleData {
+  id: string;
+  route_id: string | null;
+  trip_id: string | null;
+  shape_id: string | null;
+  stop_id: string | null;
+  direction_id: DirectionId;
+  longitude: number;
+  latitude: number;
+  bearing: number;
+  status: string;
+  crowding: CrowdingType | null;
+}
+
+export interface Vehicle {
+  id: string;
+  route_id: string | null;
+  trip_id: string | null;
+  shape_id: string | null;
+  stop_id: string | null;
+  direction_id: DirectionId;
+  longitude: number;
+  latitude: number;
+  bearing: number;
+  status: string;
+  crowding: CrowdingType | null;
+}
```

Comment:
We already have a Vehicle interface, could this be renamed something unique?  Or the old one and this one merged?

Reply:
I'm hesitant to merge them since I'm not entirely sure what the existing Vehicle is intended to represent. Since this vehicle represents a pure vehicle from the API, I'm not sure if there is an appropriate name qualifier for it.

Reply:
(non-blocking): Maybe a name like `ApiVehicle` or `VehicleApi`